src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2870,6 +2870,8 @@ public static unsafe Document ParseRawDataSectionRevisionWithValidation(JsonOper
             if (size > expectedSize || size <= 0)
                 throw new ArgumentException("Data size is invalid, possible corruption when parsing BlittableJsonReaderObject", nameof(size));
 
+            BlittableJsonReaderObject.BlittableValidation(context, ptr, size);
```

Comment:
We do it every time, not only from Voron.Recovery. Is it how it was supposed to be?